cilium/cmd/debuginfo.go

```diff
@@ -93,43 +180,85 @@ func runDebugInfo(cmd *cobra.Command, args []string) {
 		os.Exit(1)
 	}
 
-	// define output type and file path
 	var output outputType
-	var path string
-
-	switch {
-	case len(file) > 0: // Markdown file
-		output = MARKDOWN
-		path = file
-	case len(html) > 0: // HTML file
-		output = HTML
-		path = html
-	default: // Write to standard output
-		output = STDOUT
-	}
 
 	// create tab-writer to fill buffer
 	var buf bytes.Buffer
 	w := tabwriter.NewWriter(&buf, 5, 0, 3, ' ', 0)
 	p := resp.Payload
 
-	// generate multiple files
-	if (len(file) > 0 || len(html) > 0) && filePerCommand {
-		for cmdName, section := range sections {
+	cmdTime := time.Now()
+
+	if outputToFile && len(outputOpts) == 0 {
+		outputOpts = append(outputOpts, markdownOutput)
+	}
+
+	// Dump payload for each output format.
+	for _, outputOpt := range outputOpts {
+		var fileName string
+
+		switch strings.ToUpper(outputOpt) {
```

Comment:
Seems like if the validation also sanitizes, then this switch statement would be unnecessary.

Reply:
I don't follow - I did this to map from the string in `outputOpt` to the corresponding enum.

Reply:
That's sorta my point :-) I was thinking that this mapping could occur upon validation so that you don't have to map it here, you could just use the enum as-is.

Reply:
Will be fixed in next upload.